services/movie.py

```diff
@@ -0,0 +1,37 @@
+from django.db.models import QuerySet
+
+from db.models import Movie
+
+
+def get_movies(
+    genres_ids: list[int] = None,
+    actors_ids: list[int] = None
+) -> QuerySet:
+    queryset = Movie.objects.all()
+
+    if genres_ids:
+        queryset = queryset.filter(genres__id__in=genres_ids)
+    if actors_ids:
+        queryset = queryset.filter(actors__id__in=actors_ids)
+    return queryset.distinct()
+
+
+def get_movie_by_id(movie_id: int) -> Movie:
+    return Movie.objects.get(pk=movie_id)
+
+
+def create_movie(
+        movie_title: str,
+        movie_description: str,
+        genres_ids: list[int] = None,
+        actors_ids: list[int] = None
```

Comment:
list or None type

Reply:
Can you explain me please this note, I wrote None cuz it's just default value

Reply:
Right - when you assign default value as None it means that parameter is optional but it is not explicitly shown in type annotation. You should use `Optional` (old school style) or use `| None` ( as of python 3.9) on `list[int]` . Please check and adjust everywhere throughout your code

Reply:
Ok, got you, now I understand